R/HubbellSE.R

```diff
@@ -0,0 +1,122 @@
+                        row.names = rownames(paste0("species", seq_len(10))),
+                        stringsAsFactors = FALSE)
+row_data <- t(row_data)
+
+setMethod("HubbellSE", signature = c(N="numeric"),
+    function(N, M, I, d, m, tskip = 0 , tend, norm = FALSE){
+            # First setting the function arguments right
+            pbirth <- runif(N, min = 0, max = 1)
+            pmigr <- runif(M, min = 0, max = 1)
+            if(length(pbirth)!=N | length(pmigr)!=M){
+                stop("Either length of pbirth vector does not match with N or
+                    length of pmigr vector does not match with M")
+            }
+            pbirth <- c(pbirth, rep(0, times = (M-N)))
+            pbirth <- pbirth/sum(pbirth)
+            pmigr <- pmigr/sum(pmigr)
+            com <- ceiling(I*pbirth)
+            if(sum(com)<I){
+                ind <- sample(1:M, size = I-sum(com), prob = pbirth)
+                com[ind] <- com[ind] +1
+            } else if(sum(com)>I){
+                ind <- sample(1:M, size = sum(com)-I, prob = 1-pbirth)
+                com[ind] <- com[ind] -1
+            }
+            # The simulation
+            tseries <- matrix(0, nrow = M, ncol = tend)
+            colnames(tseries) <- paste0("t", 1:tend)
+            rownames(tseries) <- 1:M
+
+            com[which(com < 0)] <- 0
+            tseries[,1] <- com
+            for (t in 2:tend){
+            # Each iteration the probability of births is updated by the counts
+                pbirth <- com/sum(com)
+                pbirth[which(pbirth < 0)] <- 0
+            # Probability of births is used to pick the species that will die
+            # because species with count 0 will have probability 0 and species
+            #not present in the community can also not die
+                deaths <- rmultinom(n = 1, size = d, prob = pbirth)
+            while(sum(com-deaths <0) >0){
+                neg_sp <- which(com-deaths <0)
+                pbirth[neg_sp] <- 0
+                deaths <- rmultinom(n = 1, size = d, prob = pbirth)
+            }
+
+            event <- rbinom(d, 1, prob = m) # immigration rate m: probability
+            #death replaced by immigrant
+            #immigration 1, birth 0
+            n_migrants <- sum(event)
+            n_births <- length(event) - n_migrants
+
+            births <- rmultinom(1, n_births, prob = pbirth)
+            migr <- rmultinom(1, n_migrants, prob = pmigr)
+            com <- com - deaths + births + migr
+            com[which(com < 0)] <- 0
+            tseries[,t] <- com
+            }
+            if(norm){
+                tseries <- t(t(tseries)/colSums(tseries))
+            }
+
+            AbundanceM <- tseries[, (tskip +1):tend]
+            SE <- SummarizedExperiment(assays = list(counts = AbundanceM),
+                                                        colData = col_data,
+                                                        rowData = row_data )
```

Comment:
remove the empty space before parenthesis

Reply:
What i was thinking is that maybe i should remove colData and rowData sections and put them into example part because i arranged them according to the example abundance matrix. But i don't know what to put instead, maybe just SE <- SummarizedExperiment(assays = list(counts = AbundanceM)) is enough?

Reply:
Yes that's ok.